libp2p/protocols/pubsub/pubsubpeer.nim

```diff
@@ -478,19 +478,16 @@ iterator splitRPCMsg(
   ## exceeds the `maxSize` when trying to fit into an empty `RPCMsg`, the latter is skipped as too large to send.
   ## Every constructed `RPCMsg` is then encoded, optionally anonymized, and yielded as a sequence of bytes.
 
-  var currentRPCMsg = rpcMsg
-  currentRPCMsg.messages = newSeq[Message]()
-
-  var currentSize = byteSize(currentRPCMsg)
+  var currentRPCMsg = RPCMsg()
```

Comment:
Don't you lose the `subscriptions`, `control`, `ping` and `pong` messages that the original message might have? because according to specs, a message might have control messages piggybacked in it https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.0.md#control-message-piggybacking

Reply:
on line [495](https://github.com/vacp2p/nim-libp2p/pull/1665/files/31e233ed96e1a35a83c3e4aa661c04c6554f0b3f#diff-cd23f9cd2a0f9c037ef0b4245ee7d06c09a24985c8ee5ed39fff496f705570c3R495) after batch is sent we are already loosing that.